### PR TITLE
nakamoto updates

### PR DESCRIPTION
**Pull request by ryanwaits**
## What does this PR do?

- [x] add a banner and call to action that points to new nakamoto updates page
- [x] update `.md` extensions to `.mdx` in stacks.js module to avoid build issues
- [x] create a new nakamoto updates page with a changelog for all products
- [x] add a sidebar item for this new page
- [x] add a new api reference to `nakamoto.10` at `/nakamoto-api`

### CODE REVIEW

**Comment by ryanwaits** [comment]
in addition, ran into some odd build issues with some formatting in the stacks.js markdown files, so i reformatted a few lines and changed the extensions to use `.mdx` instead of `.md` which seems to have solved the issue.

...in case anyone is wondering why the diff is larger than expected.

**Comment by diwakergupta** [comment]
> observing an issue with our layout pages that should render banners post `docusaurus 3` version upgrade, and having trouble figuring out what the breaking change was and why they aren't showing up. for now ill commit the version back to `2.4.3` since this shouldn't break any existing updates since then

hmm, feels like a pretty big hammer (reverting back to docusaurus 2) for this one nail. can you do just the changelog and sidebar for now, I'm happy to dig into the banner issue separately. that way you needn't block on downgrading the whole thing.

**Comment by ryanwaits** [comment]
> can you do just the changelog and sidebar for now

yeah that's fine with me if we're cool without the banner

i also wouldn't drain too much energy on the banner issue in v3, i already wasted enough 😆

**Comment by ryanwaits** [comment]
my guess is the `linkcheck` is failing because of a reference to the new `/nakamoto-api` endpoint so should fix itself when merged

File: docs/nakamoto/stacks-api.mdx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+title: Nakamoto Updates for Stacks API
+---
+
+## What's New
+
+### Nakamoto endpoints
```
Comment (saralab) [inline docs/nakamoto/stacks-api.mdx:7]: Include the Nakamoto Testnet URL here? https://api.nakamoto.testnet.hiro.so/

File: docs/nakamoto/index.mdx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+# Nakamoto Updates
+
+## Stacks Explorer: What's New
+
+### Introducing a Fresh View for Blocks
+
+The Stacks Explorer now features a new way to display blocks, aligning with the Nakamoto upgrade's approach of multiple STX blocks settling into one BTC block.
+
+**[Add image of the new blocks view]**
```
Comment (andresgalante) [inline docs/nakamoto/index.mdx:15]: @ryanwaits let's use this image here

image_720

File: docs/nakamoto/index.mdx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+### Tailored Viewing Experience
+
+This new view is available on [Nakamoto Testnet](https://explorer.hiro.so/?chain=testnet&api=https://api.nakamoto.testnet.hiro.so/), accessible via the Network dropdown.
+
+#### Two Distinctive Display Modes
+
+1. **Independent Display:** Focuses on STX blocks progress.
+2. **COMING SOON: Grouped by Bitcoin Block:** Shows BTC blocks flow alongside STX blocks. (coming soon)
+
+**[Add image of the view selector]**
```
Comment (andresgalante) [inline docs/nakamoto/index.mdx:26]: We can use this image here

image_720-1

File: docs/nakamoto/index.mdx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+**[Add image of the new blocks view]**
+
+### Tailored Viewing Experience
+
+This new view is available on [Nakamoto Testnet](https://explorer.hiro.so/?chain=testnet&api=https://api.nakamoto.testnet.hiro.so/), accessible via the Network dropdown.
+
+#### Two Distinctive Display Modes
+
+1. **Independent Display:** Focuses on STX blocks progress.
+2. **COMING SOON: Grouped by Bitcoin Block:** Shows BTC blocks flow alongside STX blocks. (coming soon)
```
Comment (andresgalante) [inline docs/nakamoto/index.mdx:24]: You are mentioning "coming soon" twice

File: docs/nakamoto/index.mdx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+**[Add image of the view selector]**
+
+### Stay in the Loop with Live View
+
+Keep up-to-date with blocks in real-time with our new live view option.
+
+**[Add image of the live view selector]**
```
Comment (andresgalante) [inline docs/nakamoto/index.mdx:32]: please remove this line

**Review by andresgalante — CHANGES_REQUESTED** [review]

File: docs/nakamoto/explorer.mdx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+---
+title: Nakamoto Updates for Stacks Explorer
+---
+
+## What's New
```
Comment (andresgalante) [inline docs/nakamoto/explorer.mdx:5]: Lets make here the same changes we are doing to the index.mdx

**Review by rafaelcr — APPROVED** [review]
LGTM for the API doc changes

**Review by hugocaillard — APPROVED** [review]
LGTM 👍 
Thanks @ryanwaits!

**Review by andresgalante — APPROVED** [review]
Thanks for the changes

File: docs/nakamoto/clarinet.mdx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+As of Clarinet 2.4.0, the Devnet can run the blockchain in Nakamoto mode.
+
+Once your Devnet reaches the right block height, it will deploy the `pox-4.clar` and run in
+Epoch 2.5.
+From there, Clarinet will send `stack-stx` and `stack-extend` requests to the new PoX contract.
+In Nakamoto mode, the Devnet has 2 signers. Eventually, you can observe the
+`vote-for-aggregate-public-key` transactions that they send.
+
+Do not expect to reach Epoch 3.0 with Clarinet 2.4.0. This release is focusing on Epoch 2.4.
```
Comment (smcclellan) [inline docs/nakamoto/clarinet.mdx:15]:
```suggestion
Do not expect to reach Epoch 3.0 with Clarinet 2.4.0. This release is focusing on Epoch 2.5.
```

File: docs/nakamoto/clarinet.mdx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+Epoch 3.0, with fast blocks and others improvements, will come in a future release.
+
+### Setting up a project for Nakamoto
+
+Using one of you Clarinet projects, or a new one (with `clarinet new nakamoto-project`), update the
```
Comment (smcclellan) [inline docs/nakamoto/clarinet.mdx:20]:
```suggestion
Using one of your Clarinet projects, or a new one (with `clarinet new nakamoto-project`), update the
```

File: docs/nakamoto/clarinet.mdx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+`settings/Devnet.toml`.
+
+Under the `[devnet]` section, set `use_nakamoto = true`, the `pox-4.clar` contract should be
```
Comment (smcclellan) [inline docs/nakamoto/clarinet.mdx:23]:
```suggestion
Under the `[devnet]` section when `use_nakamoto = true` is set, the `pox-4.clar` contract should be
```

File: docs/nakamoto/index.mdx [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+
+See [Nakamoto Preview 1](https://github.com/hirosystems/clarinet/releases/tag/nakamoto-preview-1) and [Clarinet PR-1350](https://github.com/hirosystems/clarinet/pull/1350) for relevant notes + PRs.
+
+- Clarinet’s devnet can now be used to test Nakamoto functionality. To enable this, add the `use_nakamoto = true` flag to your `Devnet.toml`.
+- Enabling this feature will require accurate pox stacking orders. (See the release notes above for details)
+- The devnet now starts an additional process - the Stacks Signer. (Details on how to configure the signer address coming soon)
+
+### `clarinet check` Changes
+
+- Running clarinet check --enable-clarity-wasm now runs the current interpreter and  clarity wasm  side-by-side to allow comparing outputs.
```
Comment (smcclellan) [inline docs/nakamoto/index.mdx:62]:
```suggestion
- Running `clarinet check --enable-clarity-wasm` now runs the current interpreter and clarity wasm side-by-side to allow comparing outputs.
```

File: docs/nakamoto/index.mdx [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+
+### `clarinet console` Changes
+
+- clarinet console now automatically runs the current interpreter and clarity wasm side-by-side and logs any difference in outputs.
```
Comment (smcclellan) [inline docs/nakamoto/index.mdx:66]:
```suggestion
- `clarinet console` now automatically runs the current interpreter and clarity wasm side-by-side and logs any difference in outputs.
```

File: src/components/_content.ts [hunk 12]
```diff
@@ -21,13 +21,23 @@ export default {
...
   banner: {
     type: 'info',
-    from: '2023-11-20',
-    to: '2023-12-31',
-    text: ["We've made changes to our rate limits for Hiro APIs"],
-    cta: 'Learn more',
-    ctaLink: 'https://www.hiro.so/blog/updated-rate-limits-for-hiro-apis',
+    from: '2024-03-15',
+    to: '2024-04-20',
+    text: ['The Nakamoto upgrade is now live on testnet: see a list of Hiro product changes here.'],
```
Comment (smcclellan) [inline src/components/_content.ts:37]:
```suggestion
    text: ['The Nakamoto upgrade is now live on Testnet! See a list of Hiro product changes here.'],
```

**Review by smcclellan — APPROVED** [review]
These are formatting improvements -- otherwise looks great!

**Comment by ryanwaits** [comment]
> These are formatting improvements -- otherwise looks great!

thanks @smcclellan !

File: docs/nakamoto/explorer.mdx [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+
+### Introducing a Fresh View for Blocks
+
+The Stacks Explorer now features a new way to display blocks, aligning with the Nakamoto upgrade's approach of multiple STX blocks settling into one BTC block.
+
+![Block View](./images/block-view.png)
+
+### Tailored Viewing Experience
+
+This new view is available on [Nakamoto Testnet](https://explorer.hiro.so/?chain=testnet&api=https://api.nakamoto.testnet.hiro.so/), accessible via the Network dropdown.
```
Comment (andresgalante) [inline docs/nakamoto/explorer.mdx:15]: @ryanwaits can we remove the trailing / here please

**Review by andresgalante — CHANGES_REQUESTED** [review]

File: docs/nakamoto/index.mdx [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+
+### Introducing a Fresh View for Blocks
+
+The Stacks Explorer now features a new way to display blocks, aligning with the Nakamoto upgrade's approach of multiple STX blocks settling into one BTC block.
+
+![Block View](./images/block-view.png)
+
+### Tailored Viewing Experience
+
+This new view is available on [Nakamoto Testnet](https://explorer.hiro.so/?chain=testnet&api=https://api.nakamoto.testnet.hiro.so/), accessible via the Network dropdown.
```
Comment (andresgalante) [inline docs/nakamoto/index.mdx:19]: and here, lets remove the trailing `/`

**Review by andresgalante — APPROVED** [review]

**Pull request merged**